### ISSUE TITLE
No error message when a common user clicks one public project but he is not the user of it

### ISSUE THREAD

**Issue by supereagle** [vmware/harbor#745]
A common user, not the admin, can see the list of all public projects. 
If he is not the user of one project, no error message when he try to click the project, it just redirects another project of which he is the user. A error message about no enough permission should appear to reminder user, just like when he try to change the config of publicity without enough permission.

No error when he is the user of the project, no matter what the role he is.

**Comment by wknet123** [comment]
A common user can see public project's repositories without permission check, meanwhile other navigations buttons which he has no permissions would be hidden. That's our showing logic currently.

**Comment by supereagle** [comment]
What is the difference between the permission of a common user not in the public project and the guest user of the project?
The test result is that: 
* A common user can see the list of all public projects in the "Public Projects" tab, but he can not see the repositories of one public project which he is not the user. 
* But if he is the "Guest" user of this public project, he can see the repositories of it.

**Comment by wknet123** [comment]
It may cause redirection to project list page when common user has no project created by himself, that is a issue found, and I will fix it, thanks.

**Comment by wknet123** [comment]
Have you tried it out by checking out the 'dev' branch code? It may cause redirection when a common user signed up but has no project created for himself. I will fix this issue, thanks.

**Comment by supereagle** [comment]
Maybe this is not the root cause. There is one project created by himself, and it is public. There is no error when visits the repositories of his own public project, but error when visits the repositories of public project from other users.

When the error occurs, it sends 2 requests. The 1th request is correct and can get the list of repositories, but later it sends the 2nd request for another project. Not sure the cause of the 2nd request.

**Comment by wknet123** [comment]
Fixed by #753 on branch dev and will merge to master on the mid-September.

**Comment by supereagle** [comment]
I will verify this issue later.

**Comment by wknet123** [comment]
@supereagle how about this issue status, if it had been solved, I'll close it.

**Comment by supereagle** [comment]
Have tested the dev branch, but blocked by another issue: can not create new project as "400 Bad Request". Also tested the latest code on master branch, new project can be created.

**Comment by wknet123** [comment]
Maybe caused by the refactor of DB structure, please remove the current data volume (/data by default), and restart the project.

**Comment by supereagle** [comment]
Yes. This issue has been fixed, I will close it. Thanks.
